src/modules/flow/json/json.c

```diff
@@ -405,6 +405,22 @@ json_path_is_array_key(struct sol_str_slice slice)
     for (end_reason = SOL_JSON_LOOP_REASON_OK; \
         json_path_get_next_slice(&scanner, &key_slice, &end_reason);)
 
+static int32_t
+json_array_get_key_index(struct sol_str_slice key)
+{
+    long int index_val;
+    char *endptr;
+
+    errno = 0;
+    //TODO: use safe strtol from sol-util when available
+    index_val = strtol(key.data + 1, &endptr, 10);
+    if (endptr != key.data + key.len - 1 ||
+        index_val < 0 || errno > 0 || index_val > INT32_MAX)
```

Comment:
You should also check if the value's smaller than `INT32_MIN`. A way to do that without adding another branch, is to perform `if ((long int)(int32_t)index_val != index_val)`, and remove the `if (index_val > INT32_MAX)` test.